src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -225,13 +225,13 @@ public void appendLabel(String key, String value) {
         googleLabels.put(key, value);
     }
 
-    public ComputeEngineInstance provision(TaskListener listener, Label requiredLabel) throws IOException {
+    public ComputeEngineInstance provision(TaskListener listener, String labels) throws IOException {
```

Comment:
I'm nervous about having a single string parameter which represents multiple values. This has format assumptions baked into it, which need to be extended to any callers. Consider instead using an ImmutableList<String> param and encapsulate the string formatting assumptions as an implementation detail.

Reply:
Also would like to see a unit test which ensures no regressions.

Reply:
I used that PR a bit before it was needed for my setup, and it worked with plain labels..

Under the hood they are also setting `string` labels so its rather ok..
And we would just take string split it by ` ` and then join it by ` `..
We cannot du much more I think...

Reply:
I would rather decrease any such tight-coupling. String splitting/joining is still preferable than having to maintain assumptions about the format of data in multiple places. Making the provision method take a list makes it a much simpler method.